modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/jaxrs/extension/RelationshipsResourceImpl.java

```diff
@@ -0,0 +1,169 @@
+
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.Parameters;
+import io.swagger.v3.oas.annotations.enums.ParameterIn;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import io.swagger.v3.oas.annotations.tags.Tags;
+
+import java.net.URI;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import javax.validation.constraints.NotNull;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.UriInfo;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Luis Miguel Barcos
+ */
+@Component(
+	factory = "com.liferay.object.rest.extension.RelationshipsResource",
+	property = {"api.version=v1.0", "osgi.jaxrs.resource=true"},
+	service = RelationshipsResource.class
+)
+@Path("/v1.0")
+public class RelationshipsResourceImpl implements RelationshipsResource {
+
+	/**
+	 * Invoke this method with the command line:
+	 *
+	 *  curl http://localhost:8080/o/headless-admin-user/{currentObjectEntryId}/{objectRelationshipName} -u test@liferay.com:test
+	 */
+	@GET
+	@Override
+	@Parameters(
+		{
+			@Parameter(in = ParameterIn.PATH, name = "currentObjectEntryId"),
+			@Parameter(in = ParameterIn.PATH, name = "objectRelationshipName"),
+			@Parameter(in = ParameterIn.QUERY, name = "page"),
+			@Parameter(in = ParameterIn.QUERY, name = "pageSize")
+		}
+	)
+	@Path("/{currentObjectEntryId}/{objectRelationshipName}")
+	@Produces({"application/json", "application/xml"})
+	@Tags({@Tag(name = "ObjectEntry")})
+	public Page<ObjectEntry> getCurrentObjectEntriesObjectRelationshipNamePage(
+			@NotNull @Parameter(hidden = true)
+			@PathParam("currentObjectEntryId")
+			Long currentObjectEntryId,
+			@NotNull @Parameter(hidden = true)
+			@PathParam("objectRelationshipName")
+			String objectRelationshipName,
+			@Context Pagination pagination)
+		throws Exception {
+
+		List<ObjectRelationship> objectRelationships =
+			_objectRelationshipLocalService.getObjectRelationships(
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+		Stream<ObjectRelationship> objectRelationshipsStream =
+			objectRelationships.stream();
+
+		ObjectRelationship currentObjectRelationship =
+			objectRelationshipsStream.filter(
+				objectRelationship ->
+					!objectRelationship.isReverse() &&
+					objectRelationship.getName(
+					).equals(
+						objectRelationshipName
+					)
+			).findFirst(
+			).orElse(
+				null
+			);
+
+		ObjectDefinition currentObjectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				currentObjectRelationship.getObjectDefinitionId1());
+
+		String currentObjectDefinitionRESTContextPath =
+			currentObjectDefinition.getRESTContextPath();
+
+		URI baseURI = _uriInfo.getBaseUri();
+
+		String path = baseURI.getPath();
+
+		if (!currentObjectDefinitionRESTContextPath.equals(
+				path.split("/")[2])) {
```

Comment:
I will somehow improve the readability of this condition. Is difficult to understand what it does and I don't like splitting, accessing and comparing in the same line